torch/_logging/_internal.py

```diff
@@ -911,19 +911,26 @@ def _init_logs(log_file_name=None):
     handler: Optional[logging.Handler] = None
     if trace_file_name is not None:
         handler = logging.FileHandler(trace_file_name)
-    if handler is not None:
-        trace_log.setLevel(logging.DEBUG)
-        trace_log_handler = _track_handler(handler)
-        trace_log_handler.setFormatter(TorchLogsFormatter(trace=True))
-        trace_log.addHandler(trace_log_handler)
-
-
-class FreshFileHandler(logging.StreamHandler):
+    else:
+        # This handler may remove itself if we are not actually in an FB
+        # environment.  This allows us to defer actually initializing it until
+        # we actually need to log anything.  This is important because JK
+        # initializes a C++ singleton, which will pork our process if we
+        # subsequently fork.
+        handler = LazyFbTraceHandler()
+    # This log is ALWAYS at debug level.  We will additionally test if there
+    # are any handlers before deciding to actually call logging on this.  Do
+    # not manually call
+    trace_log.setLevel(logging.DEBUG)
```

Comment:
Just for my curiosity, we would only turn on structured trace logs (a.k.a DEBUG level) or turn off, we don't provide other granularities or levels?

Reply:
Only trace logs is debug. It has to be due how the lazy initialization works

Reply:
Make sense.